### PR TITLE
I18N: Do not mark empty strings for a translation

### PR DESCRIPTION
**Pull request by ppisar**
An empty string has a special meaning in gettext and marking such string produces invalid catalogs with a duplicate message ID entry.

Fixes: #1158

### CODE REVIEW

**Comment by ppisar** [comment]
CI fails on the duplicate msgids in translations. I will also fix the catalogs in this pull request.

**Comment by ppisar** [comment]
The removal of M_() macro use actually breaks types:

~~~~
/builddir/build/BUILD/dnf5-5.1.11/libdnf5-cli/exception.cpp: In constructor ‘libdnf5::cli::SilentCommandExitError::SilentCommandExitError(int)’:
/builddir/build/BUILD/dnf5-5.1.11/libdnf5-cli/exception.cpp:49:95: error: no matching function for call to ‘libdnf5::cli::Error::Error(const char [1])’
   49 | SilentCommandExitError::SilentCommandExitError(int exit_code) : Error(""), exit_code(exit_code) {}
~~~~

**Comment by ppisar** [comment]
I will try to use a new macro for  `{ .bgettextMsg = "\004" GETTEXT_DOMAIN "\00" "" }`  instead of `M_("")`. But I wonder whether then Error() does not print a translation for an msgid "", i.e metadata about a translation catalog.

**Comment by ppisar** [comment]
The tests starts looking good. I will put a commit on top to suppress clang-format at two places.

File: include/libdnf5/utils/bgettext/bgettext-mark-domain.h [hunk 1]
```diff
@@ -26,6 +26,9 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include "bgettext-mark-common.h"
 
+// Use EMPTY_MESSAGE instead of M_(""). Otherwise, xgettext breaks catalogs.
+#define EMPTY_MESSAGE \
+    { .bgettextMsg = "\004" GETTEXT_DOMAIN "\00" "" }
```
Comment (m-blaha) [inline include/libdnf5/utils/bgettext/bgettext-mark-domain.h:31]: Can you please fix the formatting?  See https://github.com/rpm-software-management/dnf5/actions/runs/7503905055/job/20429713582?pr=1162

Comment (ppisar) [reply include/libdnf5/utils/bgettext/bgettext-mark-domain.h:32]: Do you mean suppress clang-format, or format all the macros according to the clang-format's taste? I've just pushed a change which does the former.

Comment (m-blaha) [reply include/libdnf5/utils/bgettext/bgettext-mark-domain.h:32]: Thanks, I also find suppressing clang format better here.

**Review by m-blaha — APPROVED** [review]
ccache builds still fail for some reason, but copr builds are OK. Thank you.

**Comment by rffontenelle** [comment]
Thanks for the fix, but the existing translations in the translation repository didn't receive the fix. Weblate still fails to renew the translations files because `msgmerge` is still unable to update due to the duplicate strings present in there. https://github.com/rpm-software-management/dnf5-l10n/pull/2 should fix this for good.

**Pull request merged**